src/components/inputs/Button.tsx

```diff
@@ -40,18 +48,41 @@ const Button = (props) => {
   );
 };
 
-Button.defaultProps = {
-  id: 'button',
-  variant: 'contained',
-  color: 'primary',
-  size: 'medium',
-  margin: 2,
-  width: null,
-  startIcon: null,
-  endIcon: null,
-  nClicks: 0,
-  disabled: false
-};
+// TypeScript props type
+type ButtonProps = {
+  /** Button text */
+  text?: string;
+  /** MUI button variant */
+  variant?: 'contained' | 'outlined' | 'text';
+  /** MUI button color */
+  color?: 'primary' | 'secondary' | 'inherit' | 'default';
+  /** Icon color */
+  iconColor?: 'primary' | 'secondary' | 'action' | 'error' | 'disabled';
+  /** MUI button size, small | medium | large */
+  size?: 'small' | 'medium' | 'large';
+  /** Component margin */
+  margin?: number | string;
+  /** Button is disabled */
+  disabled?: boolean;
+  /** Button has no ripple effect */
+  disableRipple?: boolean;
+  /* Disable keyboard focus ripple */
+  disableFocusRipple?: boolean;
+  /** Disable elevation */
+  disableElevation?: boolean;
+  /** Material Icon name to display at start of button */
+  startIcon?: string;
+  /** Material Icon name to display at end of button */
+  endIcon?: string;
+  /** Button link */
+  href?: string;
+  /** Component width */
+  width?: string | null;
+  /** Number of times the button has been clicked */
+  nClicks?: number;
+} & DashComponentProps;
+
+Button.defaultProps = {};
 
 Button.propTypes = {
```

Comment:
As we define the type `ButtonProps` we can delete `Button.propTypes`. It is not present in the[ Dash TS template](https://github.com/plotly/dash-typescript-component-template/blob/master/%7B%7Bcookiecutter.project_shortname%7D%7D/src/ts/components/%7B%7Bcookiecutter.component_name%7D%7D.tsx).

Reply:
I read that after typescript builds to javascript, the runtime loses all types information, and that could be useful for Dash dev tools 🤔

Reply:
We decided to remove both `defaultProps` (since not converted to python code anyway; we do set defaults through typescript) and `propTypes` (replaced by typescript types as in ts boilerplate)